fastly/resource_fastly_service_v1.go

```diff
@@ -1,764 +1,52 @@
-		if err := readHTTPS(conn, d, s); err != nil {
-			return err
-		}
-		if err := readElasticsearch(conn, d, s); err != nil {
-			return err
-		}
-		if err := readFTP(conn, d, s); err != nil {
-			return err
-		}
-		if err := readSFTP(conn, d, s); err != nil {
-			return err
-		}
-		if err := readDatadog(conn, d, s); err != nil {
-			return err
-		}
-		if err := readLoggly(conn, d, s); err != nil {
-			return err
-		}
-		if err := readNewRelic(conn, d, s); err != nil {
-			return err
-		}
-		if err := readScalyr(conn, d, s); err != nil {
-			return err
-		}
-		if err := readGooglePubSub(conn, d, s); err != nil {
-			return err
-		}
-		if err := readKafka(conn, d, s); err != nil {
-			return err
-		}
-		if err := readResponseObject(conn, d, s); err != nil {
-			return err
-		}
-		if err := readCondition(conn, d, s); err != nil {
-			return err
-		}
-		if err := readRequestSetting(conn, d, s); err != nil {
-			return err
-		}
-		if err := readVCL(conn, d, s); err != nil {
-			return err
-		}
-		if err := readACL(conn, d, s); err != nil {
-			return err
-		}
-		if err := readSnippet(conn, d, s); err != nil {
-			return err
-		}
-		if err := readDynamicSnippet(conn, d, s); err != nil {
-			return err
-		}
-		if err := readCacheSetting(conn, d, s); err != nil {
-			return err
-		}
-		if err := readDictionary(conn, d, s); err != nil {
-			return err
-		}
-
-	} else {
-		log.Printf("[DEBUG] Active Version for Service (%s) is empty, no state to refresh", d.Id())
-	}
-
-	return nil
-}
-
-func resourceServiceV1Delete(d *schema.ResourceData, meta interface{}) error {
-	conn := meta.(*FastlyClient).conn
-
-	// Fastly will fail to delete any service with an Active Version.
-	// If `force_destroy` is given, we deactivate the active version and then send
-	// the DELETE call
-	if d.Get("force_destroy").(bool) {
-		s, err := conn.GetServiceDetails(&gofastly.GetServiceInput{
-			ID: d.Id(),
-		})
-
-		if err != nil {
-			return err
-		}
-
-		if s.ActiveVersion.Number != 0 {
-			_, err := conn.DeactivateVersion(&gofastly.DeactivateVersionInput{
-				Service: d.Id(),
-				Version: s.ActiveVersion.Number,
-			})
-			if err != nil {
-				return err
-			}
-		}
-	}
-
-	err := conn.DeleteService(&gofastly.DeleteServiceInput{
-		ID: d.Id(),
-	})
-
-	if err != nil {
-		return err
-	}
-
-	_, err = findService(d.Id(), meta)
-	if err != nil {
-		switch err {
-		// we expect no records to be found here
-		case fastlyNoServiceFoundErr:
-			d.SetId("")
-			return nil
-		default:
-			return err
-		}
-	}
-
-	// findService above returned something and nil error, but shouldn't have
-	return fmt.Errorf("[WARN] Tried deleting Service (%s), but was still found", d.Id())
-
+// Ordering is important - stored is processing order
+// Conditions need to be updated first, as they can be referenced by other
+// configuration objects (Backends, Request Headers, etc)
+var vclService = &BaseServiceDefinition{
+	Type: "vcl",
+	Attributes: []ServiceAttributeDefinition{
+		NewServiceSettings(),
```

Comment:
Looks like `Settings` are first in an ordered slice, so disregard my comments above.